packages/fast-components-react-base/src/select/select.tsx

```diff
@@ -364,13 +361,19 @@ class Select extends Foundation<SelectHandledProps, SelectUnhandledProps, Select
         if (this.props.disabled || e.defaultPrevented) {
             return;
         }
+
         switch (e.keyCode) {
             case KeyCodes.enter:
             case KeyCodes.space:
+                e.preventDefault();
                 this.toggleMenu(!this.state.isMenuOpen);
+                if (this.state.isMenuOpen && this.props.isMenuOpen !== true) {
```

Comment:
This may be a case where we actually want to use `getDerivedStateFromProps`. IE, if we get new props where it says the menu should close, we should update the state of our component to illustrate that. This would allow us to simply update the state and respond to that change here:

```
if (this.state.isMenuOpen !== true) {
    this.focusOnTrigger();
}
```
Thoughts?

Reply:
This change isn't for controlled mode, although we may need to put some thought into how we move focus in that circumstance (frankly not sure how we'd know where to move focus in that situation though).  This just focuses on the trigger in uncontrolled mode if the menu was dismissed from a keyboard event.